lib/shopify-cli/core/monorail.rb

```diff
@@ -70,29 +55,14 @@ def prompt_for_consent
           ShopifyCli::Config.set('analytics', 'enabled', opt)
         end
 
-        def monorail_payload(args:, duration:, result:)
-          {
-            cli_sha: ShopifyCli::Git.sha(dir: ShopifyCli::ROOT),
-            uname: RbConfig::CONFIG["host"],
-            args: args,
-            timestamp: Time.now.utc.iso8601(MICROSECOND_PRECISION),
-            duration: duration,
-            result: result,
-          }
-        end
-
-        def ruby_version
-          @ruby_version ||= RUBY_VERSION
-        end
-
-        def send_event(event)
-          return unless enabled?
-          return unless consented?
+        def send_event(start_time, args, err = nil)
+          return unless enabled? && consented?
```

Comment:
could this return be as early as line 19? do we gain anything from waiting all the way til here?